### ISSUE TITLE
Re: 4.7 - BillingBlock ... tpls don't get injected - ready for review

### ISSUE THREAD

**Issue by KarinG** [iATSPayments/com.iatspayments.civicrm#120]

**Comment by adixon** [comment]
What's happening here is that for payment pages with more than one payment processor enabled, the ajax method has changed in 4.7 (perhaps now using Angular?). In any case, the solution is actually to start using the core payment object method 'getDirectDebitFormFields', and maybe some other ones as well.

**Comment by adixon** [comment]
Actually, I'm kind of stuck here. I've posted to the money list for help.

**Comment by KarinG** [comment]
I've figured out how to inject the custom e.g. ACHEFT fields with CAD chq image in 4.7.1 - when triggered by a payment processor switch. However missing currency in the new (and different from the initial page load) $form - and I think we really need that esp for ACHEFT (USD vs CAD). So thinking about that now...

**Comment by KarinG** [comment]
Ok - got it! PRs coming tomorrow [for CiviCRM Core and iATS master].

**Comment by KarinG** [comment]
OK - submitted to Core: https://github.com/KarinG/civicrm-core/commit/8fa947ab13b17d48a4c9f577f9fa98f3d568af02 - JIRA - 17952 - adding currency into the AJAX request. 

This is going to require quite a bit of refactoring on the iATS Extension side of things [esp to keep things working for first page load - as well as for 4.4 and 4.6.

**Comment by KarinG** [comment]
OK first pass is in
https://github.com/iATSPayments/com.iatspayments.civicrm/commit/7882524f30f0a803dcb2ae3bf67409a7c337d62f
This requires a the CiviCRM Core commit referenced above

**Comment by KarinG** [comment]
Patch reverted - Core PR pending...

**Comment by KarinG** [comment]
Ok got this working - in master for your review: https://github.com/iATSPayments/com.iatspayments.civicrm/commit/f8ae53f0e2a63a8e77487ac782def74b17f4d687

**Comment by KarinG** [comment]
Alan - will leave this open for you to review.

**Comment by adixon** [comment]
Okay, a finished my refactoring, closing this.

There's a lingering edge case:

A page with multiple payment processors, none of which is the site default, where there is no default amount, and where the initial payment processor requires currency dependent mangling.

In that case, then the financial payment form gets loaded without a currency, so the mangling can't happen. The financial payment form is supposed to get the currency via the url, so the issue is deeper than the core patch you got in.

**Comment by tamarmeir** [comment]
Thanks, Alan! Sorry for not understanding the title of this issue as being the same as the one you just closed!

**Comment by adixon** [comment]
Yeah, no worries - your title was better, since the other one was a technical description of the solution, and yours was a user experience description.

**Comment by tamarmeir** [comment]
Hi guys,

Just tested this in 4.7.12 with iATS 1.5.4-alpha - it looks like the issue is resolved when ACH is the only payment processing option, but if credit card is selected as well, then there is still no check image or account type - if you would like me to test on semper IT to make sure that this isn't a custom code issue, let me know.

Thanks in advance for your time,
Tamar

**Comment by KarinG** [comment]
Hi Tamar - this is working on semper-it.com - I just confirmed.

**Comment by KarinG** [comment]
Hold on - I did manage to reproduce this: if the default payment processor - this happens if the default payment processor is -not- one of the payment processor options on the Contribution form.

**Comment by dinalondon** [comment]
Just checking up on this old issue.  Is there any fix?  Or is the answer that you have to have the default payment processor as one of the options on a Contribution form?
